### ISSUE TITLE
Message to self is shown twice

### ISSUE THREAD

**Issue by RalfJung** [dino/dino#591]
Steps to reproduce:
* send an unencrypted message to yourself

Actual behavior:
* the message is shown twice in the message history

Expected behavior:
* the message should be shown only once

Conversations shows the same message only once. When I send a message to myself from Conversations, it is likewise only shown once on both devices. Only sending a message from Dino is a problem.

This is using Prosody as my jabber server.

If the message is OMEMO-encrypted, instead of showing it twice, the second message it shows reads "[This message is OMEMO encrypted]".

**Comment by Perflyst** [comment]
This bug also happens with ejabberd, so it is a client bug. If any logs are needed I can provide them.

**Comment by tg--** [comment]
I have the same issue, only slightly weirder.
Dino also shows a received message twice (sent from Conversations and Gajim), but one of the copies is not OMEMO encrypted ("I sent you an OMEMO encrypted message but your client doesn’t seem to support that. Find more information on https://conversations.im/omemo"), but the second one is.

**Comment by licaon-kter** [comment]
@tg-- what version?

**Comment by tg--** [comment]
Latest HEAD (as of yesterday, 0.0~git596.20191218.c3532bd)

**Comment by licaon-kter** [comment]
Umm, so HEAD (2eb72d5dad9be99dc110417144db2f8c4446bd53):
1. 1:1 chat + OMEMO
2. write
3. close Dino
4. open Dino
5. see a new message under yours with `[This message is OMEMO encrypted]`

Not fixed or a different issue?

**Comment by RalfJung** [comment]
@licaon-kter I cannot reproduce this, even after a restart I only see every message once and no `[This message is OMEMO encrypted]`.

**Comment by licaon-kter** [comment]
@RalfJung You are compiling from HEAD or using OBS?

**Comment by RalfJung** [comment]
I am compiling from HEAD with [a custom patch applied](https://github.com/dino/dino/pull/254).

**Comment by Zash** [comment]
I suddenly see this duplication issue.

1. 1-to-1 chat with one or more contacts
2. restart
3. my own recent messages appear duplicated

Noticed after restarting for upgrade of the Debian package to `0.1.0~git34.20200324.5a98d29` (from `0.1.0~git32.20200321.42ae68e`)

Maybe relevant is that while trying to debug another issue I have deleted the Dino data directory and started over from scratch a few times.
